### PR TITLE
Added server.js

### PR DESCRIPTION
**Pull request by reald002**
Добавлены роуты с использованием mongoose.

По сути на проверку весь файл server.js :)

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const bodyParser = require('body-parser')
+// const fs = require('fs')
```
Comment (IslamRustamov1) [inline server.js]: Don't leave comments

File: server.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+const Schema = mongoose.Schema;
+
+const app = express();
+const port = 3000
```
Comment (IslamRustamov1) [inline server.js]: move this kind of data (magic strings, constant values) to .env

File: server.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    color: {
+        type: String,
+        default: "red"
+    },
+    checked: {
+        type: Boolean,
+        default: false
+    }
+});
+const Todos = mongoose.model("Todos", todosScheme)
```
Comment (IslamRustamov1) [inline server.js]: Move model related code to a separate file

File: server.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+const mongoUri = "mongodb+srv://zebelian:zebelian1@cluster0.qjcyx.mongodb.net/todolist?retryWrites=true&w=majority"
```
Comment (IslamRustamov1) [inline server.js]: Move this to .env

File: server.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+app.get('/', function(req, res) {
+    res.sendFile(__dirname + '/public/index.html');
+});
+app.get('/public/style.css', function(req, res) {
+    res.sendFile(__dirname + '/public/style.css');
+});
+app.get('/public/script.js', function(req, res) {
+    res.sendFile(__dirname + '/public/script.js');
+});
```
Comment (IslamRustamov1) [inline server.js:25]: Move serving of static content to a separate file

File: server.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    if(!changedText){
+        return res.send("Нет изменений")
+    }
+    const id = req.params.id
+    await Todos.findByIdAndUpdate(id, {
+        text: changedText
+    })
+    const data = await Todos.findById(id)
+    res.status(200).send(data)
+})
```
Comment (IslamRustamov1) [inline server.js]: Create a router for your endpoints, create a service for your business logic

File: server.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    const saveData = new Todos({
+        text: newData.text,
+        color: newData.color,
+        checked: newData.checked
+    })
+    saveData.save(function(err){
+        if(err) return console.log(err);
+        res.status(200).send(saveData)
+    });
+});
```
Comment (IslamRustamov1) [inline server.js]: Check the order of endpoints

File: server.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+app.get('/public/script.js', function(req, res) {
+    res.sendFile(__dirname + '/public/script.js');
+});
+app.get('/todos', async function(req, res) {
+    const listCollections = await mongoose.connection.db.collection("todos", async function(err, collection){
+        await collection.find({}).toArray(function(err, data){
+            res.send(data)
+        })
+    })
+});
```
Comment (IslamRustamov1) [inline server.js]: Don't use `mongoose.connection.db...`

File: server.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+    res.send(data);
+});
+app.post('/todos', jsonParser, function(req, res) {
+    const newData = req.body
+    const saveData = new Todos({
+        text: newData.text,
+        color: newData.color,
+        checked: newData.checked
+    })
+    saveData.save(function(err){
```
Comment (IslamRustamov1) [inline server.js]: Don't use callbacks

File: server.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+        if(err) return console.log(err);
+        res.status(200).send(saveData)
+    });
+});
+app.post('/todos/:id', jsonParser, async function(req, res) {
```
Comment (IslamRustamov1) [inline server.js]: Remove this endpoint

File: server.js [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+    })
+    data = await Todos.findById(id)
+    res.status(200).send(data)
+});
+app.delete('/todos/:id', jsonParser, async function(req, res) {
+    const id = req.params.id
+    const data = await Todos.findByIdAndDelete(id)
+    res.status(200).send(data)
+})
+app.patch('/todos/:id', jsonParser, async function (req, res) {
```
Comment (IslamRustamov1) [inline server.js]: Rewrite this endpoint to change all of the fields, not just text

**Comment by AnnaKulkova** [comment]
And try to stick to one style (using semicolons or not, using tabs or spaces, etc). You can use [ESlint](https://eslint.org/) to help yourself with it.

**Review by AnnaKulkova — APPROVED** [review]
LGFM

File: controllers/mainController.js [hunk 12]
```diff
@@ -0,0 +1,34 @@
+const Todos = require('../models/todo');
```
Comment (IslamRustamov1) [inline controllers/mainController.js:1]: Please, don't name controller **MainController**. If this controller is related to the todos then call it **TodoController**.

File: server.js [hunk 13]
```diff
@@ -1,67 +1,30 @@
...
-    checked: {
-        type: Boolean,
-        default: false
-    }
-});
-const Todos = mongoose.model('Todos', todosScheme);
+const router = require('./routes/router.js');
+app.use(router);
+
+// to .env /////////////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (IslamRustamov1) [inline server.js:8]: Use - **// TODO: thing that needs to be done** - kind of comments (they are being highlighted by the IDE.

**Pull request merged**